src/dune_rules/scope.ml

```diff
@@ -123,27 +135,38 @@ module DB = struct
       ()
       ~parent:(Some parent)
       ~resolve:(fun name ->
-        Memo.return
-        @@
         match Lib_name.Map.find map name with
-        | None -> Lib.DB.Resolve_result.not_found
-        | Some (Redirect lib) -> Lib.DB.Resolve_result.redirect_in_the_same_db lib
-        | Some (Found lib) -> Lib.DB.Resolve_result.found lib
+        | None -> Memo.return Lib.DB.Resolve_result.not_found
+        | Some (Redirect (lib, enabled)) ->
+          let+ enabled =
+            let+ toggle = Memo.Lazy.force enabled in
+            Toggle.enabled toggle
+          in
+          if enabled
+          then Lib.DB.Resolve_result.redirect_in_the_same_db lib
+          else Lib.DB.Resolve_result.not_found
```

Comment:
@rgrinberg mentioned that in this case we should redirect to the parent db (I can apply this change in a follow up commit)